src/Hal/Application/Extension/ExtensionService.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * (c) Jean-François Lépine <https://twitter.com/Halleck45>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Hal\Application\Extension;
+
+use Hal\Application\Config\Configuration;
+use Hal\Component\Bounds\Bounds;
+use Hal\Component\Result\ResultCollection;
+
+class ExtensionService {
+
+    /**
+     * @var Repository
+     */
+   private $repository;
+
+    /**
+     * ExtensionService constructor.
+     * @param Repository $repository
+     */
+    public function __construct(Repository $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    /**
+     * @return Repository
+     */
+    public function getRepository()
```

Comment:
I've changed to code according your other comments.

I prefer avoid methods like `getAllExtensions()`, `getFavoriteXXX()`, `getFirtXXX()`... That's useful, but, according to me, using these "shortcuts" methods increase complexity of code and transform "logical code" into "tool code".